### DISCUSSION TITLE
Play 2.9 - NoClassDefFoundError: co/typesafe/sslconfig/ssl/Ciphers

### DISCUSSION THREAD

**Discussion by yadavan88** [playframework/playframework#12164] [Q&A] [answered]
Hi, 
I know play 2.9 is not officially announced. I saw that the artifacts are available in Maven Central. I tried to apply the latest version in one of the tutorial repo am helping to manage. 
However, when I used 2.9, I got the following error while running a test. The error originating from this code:
```
private implicit def wsClient: WSClient = app.injector.instanceOf[WSClient]
```
Here is the error message:

```

[info]   com.google.inject.ProvisionException: Unable to provision, see the following errors:
[info] 
[info] 1) [Guice/ErrorInjectingConstructor]: NoClassDefFoundError: com/typesafe/sslconfig/ssl/Ciphers$
[info]   at AsyncHttpClientProvider.<init>(AhcWSModule.scala:49)
[info]   at AsyncHttpClientProvider.class(AhcWSModule.scala:48)
[info]   while locating AsyncHttpClientProvider
[info]   at AhcWSClientProvider.<init>(AhcWSModule.scala:220)
[info]       \_ for 1st parameter asyncHttpClient
[info]   at AhcWSClientProvider.class(AhcWSModule.scala:220)
[info]   while locating AhcWSClientProvider
[info]   while locating WSClient
[info] 
[info] Learn more:
[info]   https://github.com/google/guice/wiki/ERROR_INJECTING_CONSTRUCTOR
[info] 
[info] 1 error
[info] 
[info] ======================
[info] Full classname legend:
[info] ======================
[info] AhcWSClientProvider:     "play.api.libs.ws.ahc.AhcWSClientProvider"
[info] AsyncHttpClientProvider: "play.api.libs.ws.ahc.AsyncHttpClientProvider"
[info] WSClient:                "play.api.libs.ws.WSClient"
[info] ========================
[info] End of classname legend:
[info] ========================
```
I could not fix this problem, could someone help me to fix this problem?

Thanks

**Comment by mkurz** [comment]
Which tutorial repo? Is it public?

**Reply by yadavan88** [reply]
@mkurz Thanks for quick reply :) 

Yes, it is public. It is by Baeldung. The one I am trying to fix is in my fork [here](https://github.com/yadavan88/scala-tutorials/tree/play-upgrade/play-scala/application-tests). 
The test that is failing is [here](https://github.com/yadavan88/scala-tutorials/blob/play-upgrade/play-scala/application-tests/test/com/baeldung/arrival/controller/ArrivalControllerH2Test.scala).

**Reply by mkurz** [reply, marked as answer]
I fixed that for you: 

* https://github.com/yadavan88/scala-tutorials/pull/66

You need to upgrade Play to 2.9.0 and other deps also in that sub-project to make things work.
You can just merge my pull request into your branch.

BTW, I say you use `ThisBuild / libraryDependencySchemes += "org.scala-lang.modules" %% "scala-xml" % VersionScheme.Always`
you remove that after you switched to sbt 1.9.7 everywhere and upgraded Play to 2.9

**Reply by mkurz** [reply]
Quickly `grep`'ed: You need to upgrade Play in all these test projects (sbt also and all the other dependencies):

```
play-scala/rest-api/project/plugins.sbt:addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.8.2")
play-scala/play-templates/project/plugins.sbt:addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.8.2")
play-scala/play-static-assets/project/plugins.sbt:addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.8.2")
play-scala/introduction-to-play/project/plugins.sbt:addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.8.2")
play-scala/dependency-injection/project/plugins.sbt:addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.8.8")
play-scala/custom-error-handling/project/plugins.sbt:addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.8.2")
play-scala/configuration-access/project/plugins.sbt:addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.8.2")
play-scala/caching-in-play/project/plugins.sbt:addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.8.2")
play-scala/async-tasks/project/plugins.sbt:addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.8.2")
```

**Reply by yadavan88** [reply]
@mkurz That is so nice of you :) 
I really appreciate your quick help and the PR.
